crates/ty_python_semantic/src/types.rs

```diff
@@ -7492,6 +7531,42 @@ impl<'db> TypeAliasType<'db> {
     }
 }
 
+#[salsa::interned(debug)]
+pub struct BareTypeAliasType<'db> {
+    #[returns(ref)]
+    pub name: String,
+    pub value: Type<'db>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, salsa::Update)]
+pub enum TypeAliasType<'db> {
+    PEP695(PEP695TypeAliasType<'db>),
+    Bare(BareTypeAliasType<'db>),
+}
+
+impl<'db> TypeAliasType<'db> {
+    pub(crate) fn name(self, db: &'db dyn Db) -> &'db str {
+        match self {
+            TypeAliasType::PEP695(type_alias) => type_alias.name(db),
+            TypeAliasType::Bare(type_alias) => type_alias.name(db).as_str(),
+        }
+    }
+
+    pub(crate) fn definition(&self, db: &'db dyn Db) -> Option<Definition<'db>> {
+        match self {
+            TypeAliasType::PEP695(type_alias) => Some(type_alias.definition(db)),
+            TypeAliasType::Bare(_) => None,
```

Comment:
This is used for go-to-type-definition. It seems more involved to provide the original definition here, as we would need to keep track of it through various `Type` operations and the calling machinery.